arch/arm/core/cortex_m/mpu/arm_core_mpu.c

```diff
@@ -10,11 +10,99 @@
 #include <soc.h>
 
 #include <arch/arm/cortex_m/mpu/arm_core_mpu_dev.h>
+#include <linker/linker-defs.h>
 
 #define LOG_LEVEL CONFIG_MPU_LOG_LEVEL
 #include <logging/log.h>
 LOG_MODULE_REGISTER(mpu);
 
+
+#if defined(CONFIG_APP_SHARED_MEM)
+#define _MPU_DYNAMIC_REGIONS_AREA_START ((u32_t)&_app_smem_start)
+#else
+#define _MPU_DYNAMIC_REGIONS_AREA_START ((u32_t)&__kernel_ram_start)
+#endif /* CONFIG_APP_SHARED_MEM */
+#define _MPU_DYNAMIC_REGIONS_AREA_SIZE ((u32_t)&__kernel_ram_end - \
+		_MPU_DYNAMIC_REGIONS_AREA_START)
+
+/**
+ * @brief Use the HW-specific MPU driver to program
+ *        the static MPU regions.
+ *
+ * Program the static MPU regions using the HW-specific MPU driver. The
+ * function is meant to be invoked only once upon system initialization.
+ *
+ * If the function attempts to configure a number of regions beyond the
+ * MPU HW limitations, the system behavior will be undefined.
+ *
+ * For some MPU architectures, such as the unmodified ARMv8-M MPU,
+ * the function must execute with MPU enabled.
+ */
+void _arch_configure_static_mpu_regions(void)
+{
+	/* Define a constant array of k_mem_partition objects
+	 * to hold the configuration of the respective static
+	 * MPU regions.
+	 */
+	const struct k_mem_partition static_regions[] = {
+#if defined(CONFIG_APPLICATION_MEMORY)
+		{
+		.start = (u32_t)&__app_ram_start,
+		.size = (u32_t)&__app_ram_end - (u32_t)&__app_ram_start,
+		.attr = K_MEM_PARTITION_P_RW_U_RW,
+		},
+#endif /* CONFIG_APPLICATION_MEMORY */
+#if defined(CONFIG_COVERAGE_GCOV) && defined(CONFIG_USERSPACE)
+		{
+		.start = (u32_t)&__gcov_bss_start,
+		.size = (u32_t)&__gcov_bss_end - (u32_t)&__gcov_bss_start,
+		.attr = K_MEM_PARTITION_P_RW_U_RW,
+		},
+#endif /* CONFIG_COVERAGE_GCOV && CONFIG_USERSPACE */
+#if defined(CONFIG_NOCACHE_MEMORY)
+		{
+		.start = (u32_t)&_nocache_ram_start,
+		.size = (u32_t)&_nocache_ram_end - (u32_t)&_nocache_ram_start,
+		.attr = K_MEM_PARTITION_P_RW_U_RW_NOCACHE,
```

Comment:
OTOH, this one looks good. This probably means we want to introduce `K_MEM_PARTITION_P_RW_U_RW_CACHE`. Alternatively the oring might be done in this file instead of adding too many defines.

Reply:
@aurel32 _CACHE is  (I assume) by default, pls consult the public arm_mpu.vx.h headers :)